### ISSUE TITLE
Poor coverage of the pareto front with qNEHVI

### ISSUE THREAD

**Issue by fernexda** [facebook/Ax#1785]
Dear all,
first of all, thank you for this amazing package that's been very helpful recently.
I am running aerodynamic experiments in a water channel, and the fast convergence of MOBO helps me reduce the experimental cost. It has worked well in some cases, but for another case, with more challenging (=conflicting) objectives, the MOO does not produce a satisfying pareto front.

I am running MOO following more or less the qNEHVI settings described [here](https://ax.dev/tutorials/multiobjective_optimization.html). I have 8 RangeParameters, and I'm using the qNEHVI acquisition function since it's experimental work with inherent noise. I have estimated the standard deviations of both objectives and set them with `noise_sd`. The first objective **Obj. 1** is a distance between a measured quantity and a reference value (`Obj. 1=|x-x_ref|`) which I'm trying to minimize, so this objective should tend to zero, and negative values are impossible. The second objective is a quantity I'm trying to reduce, the smaller the better.

## Exp. 1
A first optimization run gave the following Pareto front. The red dot indicates the threshold. The points are largely clustered around values of Obj. 2 between 0.5 and 1, where they don't contribute to the hypervolume improvement. The whole bottom right part remains unexplored.
![git_1](https://github.com/facebook/Ax/assets/7616916/14faec9d-bc02-496b-8aa6-53811b042190)

## Exp. 2 - better threshold
To improve the pareto front, I've changed the threshold to a value closer to the PF (inferred from Exp. 1). The result can be seen here:
![git_2](https://github.com/facebook/Ax/assets/7616916/dcead2e7-856e-415a-a37e-f187e5fe4a34)
And a zoom in with the PF here:
![git_2_zoom](https://github.com/facebook/Ax/assets/7616916/34e759a7-c163-4c07-a057-8015e323916d)

As the plot shows, the MOO focuses this time on the bottom right corner, and produces a very sparse pareto front.

To obtain a denser pareto front, I have used the `compute_posterior_pareto_frontier` function, and obtained the black circles in this image. The posterior pareto frontier performs significantly better than the one obtained during optimization, which I find surprising. I was really expecting the optimization to cover more broadly the pareto front.
![git_3](https://github.com/facebook/Ax/assets/7616916/5b47bfb0-2fe6-45b0-9a4b-a4a30ddce618)

## Comparison Exp. 1 and Exp. 2
Even more surprising, the parameters corresponding to the pareto fronts from Exp. 1 and Exp. 2 are significantly different. And the PF themselves are also different. On this figure, I've added the PF from Exp. 1, and it's clearly better than that obtained after Exp. 2. 
![git_4](https://github.com/facebook/Ax/assets/7616916/0d7ef9da-9b3c-46b1-840c-14309664d68a)

## Questions
In conclusion, I have two problems:
- the qNEHVI does not find a good coverage of the pareto front
- the pareto front changes significantly between two experiments, with very different optimal parameters in each case.

I suspect that the overall problem is a lack of exploration from the algorithm.  It seems like it gets stuck in a local minimum and stops exploring elsewhere. Any idea about how I could solve this issue? Are there any parameters to boost the exploration? I have seen similar issues [here](https://github.com/facebook/Ax/issues/879) and [here](https://github.com/facebook/Ax/issues/800), but nothing that really solved my problem.

Any help will be greatly appreciated!

**Comment by fernexda** [comment]
Hello @sdaulton,
thanks for the quick reply!

For the noise, I have selected a set of parameters, and repeated the experiment a few times, with each time slightly different results objectives. Then I've computed the standard deviation of these collected objectives. Is this correct? In this experiment, it's safe to assume that the noise level is roughly the same for a different set of parameters (is it what homoskedastic means?).

Checking the cross-validation was a good idea, here are the plot for Obj. 1 (MaintainCp)
![newplot(3)](https://github.com/facebook/Ax/assets/7616916/9a1debb9-fd88-4d64-8353-48a28b67abd1)
and Obj. 2 (ReduceCthrust):
![newplot(2)](https://github.com/facebook/Ax/assets/7616916/db5a599b-d0c1-4f56-97ef-eced23b8107a)
I guess the cross-validation is not really encouraging... Could this be the root of the problem? You think I should try and optimize the GP to fit this dataset?

Regarding the GP variable, is it possible to minimize minimize my objective `|x-x_ref|` by modeling only `x` with the GP?

I was wondering: since my Obj. 1 is not really a minimization problem (not trying to get Obj. 1 small, but maintain it to zero), could this be messing with the algorithm? I was considering going for a *single* objective optimization for Obj. 2, and adding something like  `Obj 1.<1.10^-3` as an OutcomeConstraints. Do you think that would make sense?

**Comment by fernexda** [comment]
Just for information, I've re-run the model, using `Models.FULLYBAYESIANMOO` instead of the usual `get_MOO_EHVI`. It takes significantly longer, but the results look a bit better, although still far from perfect. Obj.1:
![newplot(4)](https://github.com/facebook/Ax/assets/7616916/e61ce276-05a1-460c-9464-59514533538e)
Obj. 2:
![newplot(5)](https://github.com/facebook/Ax/assets/7616916/811a6e03-2771-4988-99b0-efa6b436a033)

**Comment by fernexda** [comment]
By rechecking the cross-validation plots, I realized that the plots of the most visible points on the plots are actually from the original random set (generated with sobol), where I have only sparse data, and the uncertainty is high. The region of interest is in fact the bottom right in both figures, where most points agglomerate (and which corresponds to the higher density of points on the pareto front image). Here is a zoom in for both figures of the region of interest.
![newplot(8)](https://github.com/facebook/Ax/assets/7616916/f1dbee3b-9db9-4685-bcc5-f73927675ca8)
![newplot(6)](https://github.com/facebook/Ax/assets/7616916/505f80a2-35c5-4cb9-aa50-9f3da3d067c1)
In this region, where the optimization seems to get stuck, the model fit looks reasonable.

**Comment by Balandat** [comment]
> In this region, where the optimization seems to get stuck, the model fit looks reasonable.

Seems like the model is reasonably well calibrated, but it's having a hard time predicting anything but the (local) mean of the results. Are your observations very noisy by any chance? Seems like the model is able to find the optimal region but then the signal-to-noise ratio becomes too bad to make more progress (which may actually be ok, depending on the problem).

**Comment by fernexda** [comment]
Hello @Balandat, thank you for your feedback. I've verified the noise level for various sets of parameters, and it remains roughly the same as the value I have given at the beginning with `noise_sd`, so I hoped this wouldn't be the problem. And in the first experiment, the PF is better than that of the second experiment by a magnitude which exceeds the noise level.

Do you have any opinion on this?
> I was wondering: since my Obj. 1 is not really a minimization problem (not trying to get Obj. 1 small, but maintain it to zero), could this be messing with the algorithm? I was considering going for a single objective optimization for Obj. 2, and adding something like `Obj 1.<1.10^-3` as an OutcomeConstraints. Do you think that would make sense?

**Comment by bernardbeckerman** [comment]
@fernexda it looks like the reference point in your first plot is close to 0.1 as well (maybe a bit higher - 0.15?). I wonder if making it much bigger (e.g., 0.4 or 0.6 - as long as those values are still of interest to you) might help. Also, I'm not sure if you're familiar for the guidelines for setting the reference point, but the rule of thumb that @sdaulton taught me is that each objective threshold should be set to be slightly worse than its "disqualification value", i.e., the objective value that is just poor enough to disqualify any arm exhibiting that value from being a usable candidate. Does that make sense?

**Comment by sdaulton** [comment]
> For the noise, I have selected a set of parameters, and repeated the experiment a few times, with each time slightly different results objectives. Then I've computed the standard deviation of these collected objectives. Is this correct? In this experiment, it's safe to assume that the noise level is roughly the same for a different set of parameters (is it what homoskedastic means?).

That makes sense, but the noise estimate might not be great if the experiment is only repeated a few times. You believe the noise level is the same across the search space (homoskedastic), then you might see better performance from using an inferred noise level by specifying the noise at each observation as `float("nan")`.

> @sdaulton I think @fernexda doesn't want to minimize or maximize x-x_ref, but make it closer to 0. Is it possible to model x-x_ref (which to your point is better suited to a normal distribution) but then perform acquisition on |x-x_ref|

In botorch this is trivial, but this is not easily configured in Ax at the moment.

> I was wondering: since my Obj. 1 is not really a minimization problem (not trying to get Obj. 1 small, but maintain it to zero), could this be messing with the algorithm? I was considering going for a single objective optimization for Obj. 2, and adding something like Obj 1.<1.10^-3 as an OutcomeConstraints. Do you think that would make sense?

This is likely a good approach. Specifically, you could add two outcome constraints on `x` such that `x_ref+EPS > x` and `x_ref-EPS < x`. Then `x` can be the metric modeled, which might help with the model fits due to better representation of the noise.

**Comment by fernexda** [comment]
Thanks @sdaulton for the feedback. I had the feeling, when I originally played with the noise, that ax needs to run additional evaluations of the objective functions when the noise is inferred (with `float("nan")`). Is that correct? Something like estimating the noise by rerunning with the same parameters? It would slightly influence my experimental workflow if that's the case.

Does an outcome constraint need to be defined on a modeled metric? Because in my case, I'm running an experiment, from which I measure two independent physical quantities, `x` and `y`, which define the two objectives, s.t. `Obj1=|x-xref|` and `Obj2 = y`. In a double-objective optimization, both metrics are modeled with GPs. If I turn this into a single-objective optimization, I will have `Obj2` from `y` as a metric, modeled with a GP, and I would like the outcome constraint on `Obj1` from `x`, which will *not* be modeled by a GP. Is this problematic? Should the constraint be applied to a modeled objective?

**Comment by sdaulton** [comment]
> I had the feeling, when I originally played with the noise, that ax needs to run additional evaluations of the objective functions when the noise is inferred (with float("nan")). Is that correct? Something like estimating the noise by rerunning with the same parameters? It would slightly influence my experimental workflow if that's the case.

No Ax won't run additional evaluation with your workflow. It will infer the noise level along with the other GP hyperparamters by maximizing GP marginal likelihood (i.e. inferring the noise level will just be part of model fitting). 

> Does an outcome constraint need to be defined on a modeled metric? 

Yes, why would Obj1 not be modeled with a GP? If you define two metrics Obj1 and Obj2 and set Obj2 as the objective and Obj1 as an outcome constraint, then Ax will automatically model both metrics with GPs.

**Comment by lena-kashtelyan** [comment]
Closing this issue as inactive. @fernexda , if you still need help, please reopen it and follow up. We're unlikely to see your follow up if you don't reopen : )
